File: Easy/Others/ValidParantheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<Character>();
        for(int i=0;i<s.length();i++)
        {
        	if(isopeningTag(s.charAt(i)))
        	{
        		stk.push(s.charAt(i));
        	}
        	else if(isclosingTag(s.charAt(i)))
        	{
        		if(!stk.empty() && getclosingTag(stk.peek())==s.charAt(i))
        		{
        			stk.pop();
        		}
        		else
        		{
        			return false;
        		}
        	}
        }
        return stk.empty();
    }
    private char getclosingTag(char c)
    {
    	if(c=='(')
    	{
    		return ')';
    	}
    	if(c=='{')
    	{
    		return '}';
    	}
    	if(c=='[')
    	{
    		return ']';
    	}
    	return '-';
    }
    private boolean isopeningTag(char c)
    {
    	if(c=='{'||c=='('||c=='[')
    	{
    		return true;
    	}
    	else
    	{
    		return false;
    	}
    }
    private boolean isclosingTag(char c)
    {
    	if(c=='}'||c==')'||c==']')
    	{
    		return true;
    	}
    	else
    	{
    		return false;
    	}
    }
}

File: CSharp/Easy/Other/ValidParantheses.cs
/*
Example 1:

Input: "()"
Output: true
Example 2:

Input: "()[]{}"
Output: true
Example 3:

Input: "(]"
Output: false
Example 4:

Input: "([)]"
Output: false
Example 5:

Input: "{[]}"
Output: true
*/
public class Solution {
    public bool IsValid(string s) {
        Stack<char> stck = new Stack<char>();
        for(int i=0;i<s.Length;i++)
        {
        	if(s[i]=='('||s[i]=='['||s[i]=='{')
        	{
        		stck.Push(s[i]);
        	}
        	else if(stck.Count>0)
        	{
        		if(s[i]==')')
        		{
        			if(stck.Peek()!='(')
        			{
        				return false;
        			}
        			else
        			{
        				stck.Pop();
        			}
        		}
        		else if(s[i]==']')
        		{
        			if(stck.Peek()!='[')
        			{
        				return false;
        			}
        			else
        			{
        				stck.Pop();
        			}
        		}
        		else if(s[i]=='}')
        		{
        			if(stck.Peek()!='{')
        			{
        				return false;
        			}
        			else
        			{
        				stck.Pop();
        			}
        		}
        	}
        	else
        	{
        		return false;
        	}
        }
        return stck.Count==0;
    }
}